app/src/main/java/com/google/android/gms/samples/vision/ocrreader/aadhaar/SavedAadhaarCardActivity.java

```diff
@@ -0,0 +1,140 @@
+package com.google.android.gms.samples.vision.ocrreader.aadhaar;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.ArrayAdapter;
+import android.widget.ListView;
+import android.widget.TextView;
+
+import com.google.android.gms.samples.vision.ocrreader.R;
+import com.google.android.gms.samples.vision.ocrreader.aadhaar.utils.CardListAdapter;
+import com.google.android.gms.samples.vision.ocrreader.aadhaar.utils.DataAttributes;
+import com.google.android.gms.samples.vision.ocrreader.aadhaar.utils.Storage;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+
+public class SavedAadhaarCardActivity extends AppCompatActivity {
+
+    private ListView lv_saved_card_list;
+    private TextView tv_no_saved_card;
+    private Storage storage;
+    private JSONArray storageDataArray;
+    private ArrayList<JSONObject> cardDataList;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
```

Comment:
This is more a generic comment but we would use MVP everywhere just so we separate the view form the logic that presents it. And the data submission, processing, saving and reading will be in another distinct enity